mautic/mautic
# New segment filters

| Q  | A
| --- | ---
| Bug fix? | n
| New feature? | y
| Related user documentation PR URL | n
| Related developer documentation PR URL | n
| Issues addressed (#s or URLs) | #3067
| BC breaks? | n
| Deprecations? | n

#### Description:

This PR added new filters. 
- Visited URL Date - best use case is with Visited URL filter  If I want filter contacts just from some period
- Visited URL Count - filter contacts based on Page Hit Count
- Referer - same like Visited URL  just based on Referer
- URL title - same like Visited URL  just based on URL title
- Source - same like Visited URL  just based on source (email, forms, campaign, dynamic content)
- Email Read Date - best use case is with Email Read filter  If I want filter contacts just from some period
- Email Read Count - filter contacts based on email open_counts
- Notifications - filter for segments by notifications state based on push_leads table
- Email sent - contact list from sent email(s)
- Email URL, Redirect URL, Landing page URL check
- Sessions - visits with 30 minutes between  return - same metric like Google Analytics

#### Steps to test this PR:
1. Apply PR, Go to segments and create new segments
2.  Use Visited URL and Visited URL Date filter combination
3. Call mautic:segments:rebuild and check rebuilded database
4. Create another segment with Referer filter
5. Call mautic:segments:rebuild and check rebuilded database
6. Create another segment with Email Read  and Email Read Date filter
7. Call mautic:segments:rebuild and check rebuilded database
6. Create another segment with Notification filter - choose empty (hasn't notifications) or notEmpty (has notifcation)
7. Call mautic:segments:rebuild and check rebuilded database
8. Create another segment with Visited URL Count 
9. Call mautic:segments:rebuild and check rebuilded database
10. Create another segment with  Email Read Count
11. Call mautic:segments:rebuild and check rebuilded database
